src/Features/CSharp/Portable/MoveToResx/CSharpMoveToResxCodeFixProvider.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Composition;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Text;
+using System.Collections.Immutable;
+using System.IO;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Microsoft.CodeAnalysis.CSharp.MoveToResx
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(CSharpMoveToResxCodeFixProvider)), Shared]
+    internal class CSharpMoveToResxCodeFixProvider : CodeFixProvider
+    {
+        [ImportingConstructor]
+        public CSharpMoveToResxCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(MoveToResx.DiagnosticId);
+
+        public override FixAllProvider GetFixAllProvider()
+            => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics[0];
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
```

Comment:
either do `GetRequiredSyntaxRootAsync` or use our existing helper, which is called something like `context.TryGetRelevantNodeAsync<LiteralExpressionSyntax>()`